platform/ios/src/MGLCameraChange.h

```diff
@@ -0,0 +1,35 @@
+#import "MGLFoundation.h"
+
+typedef NS_OPTIONS(NSUInteger, MGLCameraChangeReason)
+{
+    /// No reason for the camera move is specified.
+    MGLCameraChangeReasonNone = 0,
+
+    /// Set when a public API that moves the camera is called. This may be set for some gestures
+    MGLCameraChangeReasonProgrammatic = 1 << 0,
+
+    /// User tapped the compass to reset to North
+    MGLCameraChangeReasonResetNorth = 1 << 1,// Tap on compass
+
+    /// User panned the map.
+    MGLCameraChangeReasonGesturePan = 1 << 2,
+
+    /// User pinched to zoom in/out
+    MGLCameraChangeReasonGesturePinch = 1 << 3,
```

Comment:
How important is it to distinguish a pinch that zooms in from a double-tap that zooms in, or a pinch (spread) that zooms out from a two-finger tap that zooms out? Likewise with the “quick zoom” gesture.